src/Spice86.Core/Emulator/Devices/Sound/PCSpeaker/PcSpeaker.cs

```diff
@@ -71,42 +80,43 @@ public PcSpeaker(SoftwareMixer softwareMixer, State state, IOPortDispatcher ioPo
     /// Invoked when the speaker has been turned off.
     /// </summary>
     private void SpeakerDisabled() {
+        if (_loggerService.IsEnabled(Serilog.Events.LogEventLevel.Verbose)) {
+            _loggerService.Verbose("PC Speaker disabled");
+        }
         EnqueueCurrentNote();
         _currentPeriod = 0;
     }
-    
+
     /// <summary>
     /// Invoked when the frequency has changed.
     /// </summary>
     /// <param name="source">Source of the event.</param>
     /// <param name="e">Unused EventArgs instance.</param>
-    private void FrequencyChanged(object? source, EventArgs e) {
+    private void OnFrequencyChange(object? source, EventArgs e) {
```

Comment:
```suggestion
    private void OnFrequencyChanged(object? source, EventArgs e) {
```

Reply:
After looking deeper into this, it looks like most of the counter settings can affect sound.
I changed it to OnTimerSettingChanged and did the same for the event Counter side.